code-challenges.test.js

```diff
@@ -15,35 +15,127 @@
+// Set the parameter as numericalArray.
+// Use .sort method to sort array from ascending order by specifying comparison function correctly. 
+// Return desired output. 
+
+// const minToMax= (numericalArray) => {
+//     numericalArray.sorts(function(a,b){return a-b;})
+//     return a - b; 
+// }
+
+//* RAN TEST (RESULTS BELOW): *//
+
+// FAIL 
+// TypeError: numericalArray.sorts is not a function. 
+
+// ** NOTES TO INSTRUCTOR **: I was a bit surprised to see this error as a result of running my test, I thought my logic was...well logical. What did I do incorrectly? (Hence why my code is commented out here. )
 
 // --------------------3) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator to pass in a dynamic number of arguments.
 
 // a) Create a test with an expect statement using the variables provided.
 
-const testArray1 = [3, 7, 10, 5, 4, 3, 3]
-const testArray2 = [7, 8, 2, 3, 1, 5, 4]
-// Expected output: [3, 7, 10, 5, 4, 8, 2, 1]
+describe("arrayAlliance", () => {
+    it("takes in two arrays as arguments and returns one array with no duplicate values", () => {
+        const testArray1 = [3, 7, 10, 5, 4, 3, 3]
+        const testArray2 = [7, 8, 2, 3, 1, 5, 4]
+        // Expected output: [3, 7, 10, 5, 4, 8, 2, 1]
+        expect(arrayAlliance([3, 7, 10, 5, 4, 3, 3], [7, 8, 2, 3, 1, 5, 4])).toEqual([3, 7, 10, 5, 4, 8, 2, 1])
+      
+    })
+})
 
+//* RAN TEST (RESULTS BELOW): *//
+
+// ReferenceError: arrayAlliance is not defined
 
 // b) Create the function that makes the test pass.
+
+//Goal: Takes in two arrays as arguments and returns one array with no duplicate values
+
+//Pseudocode:
+// Declare a function called arrayAlliance. 
+// Set the parameter as an array.
+// Use .concat to combine original arrays into one
+// Use .filter & indexOf to prevent duplicate values within new array 
+// Return expected output. 
+
+const arrayAlliance= (combinedArray) => {
+    combinedArray.concat() 
+    return combinedArray.filter(value) + combinedArray.indexOf(value)
+}
+//* RAN TEST (RESULTS BELOW): *//
+
+// ReferenceError: value is not defined
+
+// ** NOTES TO INSTRUCTOR **: I'm having some trouble understanding the difference between a Jest function and just writing out normal code in a console; for instance...in this problem, I know I didn't do the `.concat` correct because I didn't plug in or reference which arrays to combine and I know the error from the test I am getting is because I didn't previous define "value", but at the same time I'm not sure how to do so within the mechanisms of Jest, it's like I can write it out in a console and combine the arrays but I'm not sure how to get a passing test with the Jest parameters/above describe method information. Any insights or suggestions to get more clarity on this? I feel like I'm getting better but not really at the same time with these test, if any of this makes sense....
```

Comment:
You are getting better!  Your test is solid!  You're thinking logically.  I think you're struggling with writing the function that passes the test.  I am happy to take some time to meet with you one on one to help you. Just let me know!

Reply:
Absolutely! Thank you.